Description: Fix build failures with g++ 4.7
Author: Sebastian Ramacher <s.ramacher@gmx.at>
Last-Update: 2012-05-13
Bug-Debian: http://bugs.debian.org/672014
--- a/source/shared_lib/sources/util/util.cpp
+++ b/source/shared_lib/sources/util/util.cpp
@@ -33,6 +33,7 @@
 #ifndef WIN32
 #include <errno.h>
 #endif
+#include <unistd.h>
 
 #include "leak_dumper.h"
 
--- a/source/shared_lib/include/graphics/math_util.h
+++ b/source/shared_lib/include/graphics/math_util.h
@@ -184,10 +184,10 @@
 
 	Rect2<T> computeBoundingRect() const{
 		return Rect2i(
-			min(p[0].x, p[1].x), 
-			min(p[0].y, p[2].y), 
-			max(p[2].x, p[3].x), 
-			max(p[1].y, p[3].y));
+			std::min(p[0].x, p[1].x), 
+			std::min(p[0].y, p[2].y), 
+			std::max(p[2].x, p[3].x), 
+			std::max(p[1].y, p[3].y));
 	}
 
 	bool isInside(const Vec2<T> &pt) const{
